src/engine/GroupBy.cpp

```diff
@@ -708,8 +706,12 @@ bool GroupBy::computeOptimizedGroupByIfPossible(IdTable* result) {
 }
 
 // _____________________________________________________________________________
-std::optional<GroupBy::ExplicitlySortedData> GroupBy::checkIfExplicitlySorted(
-    std::vector<Aggregate> aggregates) {
+std::optional<GroupBy::HashMapOptimizationData>
+GroupBy::checkIfHashMapOptimizationPossible(std::vector<Aggregate> aggregates) {
```

Comment:
Does sonarcloud really not complain about the argument not being `const std::vector<...>&` (no need to copy here).
Although this is not performance critical, you should get used to the various ways of parameter passing in C++